src/home/home_screen.rs

```diff
@@ -3,32 +3,250 @@ use makepad_widgets::*;
+                direction: MessageDirection::Incoming,
+                is_read: false,
+                timestamp: "thursday".to_string(),
+            },
+            ChatPreview {
+                username: "Julian Montes de Oca".to_string(),
+                latest_message: MessagePreview::Video,
+                direction: MessageDirection::Outgoing,
+                is_read: false,
+                timestamp: "14:05".to_string(),
+            },
+            ChatPreview {
+                username: "Edward Tan".to_string(),
+                latest_message: MessagePreview::Text("thanks ed, see you there.".to_string()),
+                direction: MessageDirection::Incoming,
+                is_read: false,
+                timestamp: "yesterday".to_string(),
+            },
+            ChatPreview {
+                username: "WeChat Team".to_string(),
+                latest_message: MessagePreview::Text("Hi!".to_string()),
+                direction: MessageDirection::Incoming,
+                is_read: false,
+                timestamp: "18:45".to_string(),
+            },
+        ];
+    }
+}
+
+impl Widget for Chats {
+    fn handle_widget_event_with(
+        &mut self,
+        cx: &mut Cx,
+        event: &Event,
+        dispatch_action: &mut dyn FnMut(&mut Cx, WidgetActionItem),
+    ) {
+        let _actions = self.list_view.handle_widget_event(cx, event);
+
+        for action in _actions {
+            dispatch_action(cx, action);
         }
     }
+
+    fn get_walk(&self) -> Walk {
+        self.walk
+    }
+
+    fn redraw(&mut self, cx: &mut Cx) {
+        self.list_view.redraw(cx)
+    }
+
+    fn draw_walk_widget(&mut self, cx: &mut Cx2d, walk: Walk) -> WidgetDraw {
+        self.draw_walk(cx, walk);
+        WidgetDraw::done()
+    }
+}
+
+impl Chats {
+    pub fn draw_walk(&mut self, cx: &mut Cx2d, walk: Walk) {
+        // todo: sort by newest incoming?
+        cx.begin_turtle(walk, self.layout);
+        self.list_view
+            .set_item_range(0, self.chat_list.len() as u64, 1);
+
+        while self.list_view.draw_widget(cx).hook_widget().is_some() {
+            while let Some(item_id) = self.list_view.next_visible_item(cx) {
+                let template = match item_id {
+                    0 => id!(search_bar),
+                    _ => id!(chat),
+                };
+                let item = self.list_view.get_item(cx, item_id, template).unwrap();
+
+                item.get_label(id!(preview.username))
+                    .set_label(&self.chat_list[item_id as usize].username);
+                item.get_label(id!(preview.content))
+                    .set_label(self.chat_list[item_id as usize].latest_message.text());
+                item.get_label(id!(timestamp))
+                    .set_label(&self.chat_list[item_id as usize].timestamp);
+
+                item.draw_widget_all(cx);
+            }
+        }
+
+        cx.end_turtle();
+    }
+}
+
+pub struct ChatPreview {
```

Comment:
This struct has the same name that the ui element defined in the live_design. What about using different identifiers? Maybe `ChatPreviewData`, `ChatPreviewInfo` or `ChatEntry`?

Reply:
Good catch! renamed to `ChatEntry`